File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    days = []
    n = len(progresses)
    for i in range(n-1, -1, -1):
        p = progresses[i]
        s = speeds[i]
        d = math.ceil((100 - p) / s)
        days.append(d)
    
    result = []
    while days:
        d = days.pop()
        cnt = 1
        while days and days[len(days)-1] <= d:
            days.pop()
            cnt += 1
        result.append(cnt)
        
    return result
            
        

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        // 작업 일수 배열
        int len = progresses.length;
        int[] days = new int[len];
        for (int i = 0; i < len; i++) {
            days[i] = (int)Math.ceil((double)(100 - progresses[i]) / speeds[i]);
        }
        
        int p = 0;
        int pending = 0;
        
        List<Integer> result = new ArrayList<>();
        for (int q = 1; q < len; q++) {
            if (days[p] < days[q]) {
                result.add(pending + 1);
                pending = 0;
                p = q;
                continue;
            }
            pending++;
        }
        result.add(pending+1);
        
        int[] arr = new int[result.size()];
        for (int i = 0; i < result.size(); i++){
            arr[i] = result.get(i);
        }
        return arr;
    }
}